serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/web/AdminController.java

```diff
@@ -0,0 +1,152 @@
+            }
+            return ResponseEntity.notFound().build();
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+    }
+
+    @GetMapping(value = "/processes", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<Process>> getProcesses() {
+        try {
+            return ResponseEntity.ok(processService.findAll());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+        }
+    }
+
+    @PutMapping(value = "/processes/{processIdentifier:.+}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity updateProcess(@PathVariable String processIdentifier, @RequestBody Process processWithValuesForUpdate) {
+        try {
+            if (processService.update(processIdentifier, processWithValuesForUpdate)) {
+                return ResponseEntity.ok().build();
+            } else {
+                return ResponseEntity.notFound().build();
+            }
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+    }
+
+
+    @GetMapping(value = "/documentTypes/{identifier}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<DocumentType> getDocumentType(@PathVariable String identifier) {
+        Optional<DocumentType> documentType = documentTypeService.findByIdentifier(identifier);
+        return documentType.map(ResponseEntity::ok).orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    @PostMapping(value = "/documentTypes", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity addDocumentType(@RequestBody DocumentType documentType) {
+        try {
+            Optional<DocumentType> existingDocumentType = documentTypeService.findByIdentifier(documentType.getIdentifier());
+            if (existingDocumentType.isPresent()) {
+                return ResponseEntity.status(HttpStatus.CONFLICT).build();
+            }
+            documentTypeService.add(documentType);
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString("/documentType")
+                    .path(documentType.getIdentifier())
+                    .build();
+            return ResponseEntity.created(uriComponents.toUri()).build();
+        } catch (Exception e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+    }
+
+    @DeleteMapping(value = "/documentTypes/{identifier:.+}")
+    public ResponseEntity deleteDocumentType(@PathVariable String identifier) {
+        try {
+            Optional<DocumentType> documentType = documentTypeService.findByIdentifier(identifier);
+            if (documentType.isPresent()) {
+                documentTypeService.delete(documentType.get());
+                return ResponseEntity.noContent().build();
+            }
+            return ResponseEntity.notFound().build();
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)